File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/fileexplorer/SpBean.java
package com.didichuxing.doraemonkit.kit.fileexplorer;

import com.didichuxing.doraemonkit.constant.SpInputType;

public class SpBean {
    public String key;
    public Object value;
    public Class clazz;

    private SpBean() {

    }

    public SpBean(String key, Object value) {
        this.key = key;
        this.value = value;
        clazz = value.getClass();
    }

    public Object toDefaultClass(String string) {
        setDefaultClass(string);
        return value;
    }

    private void setDefaultClass(String string) {
        switch (clazz.getSimpleName()) {
            case SpInputType.FLOAT:
                value = Float.valueOf(string);
                break;
            case SpInputType.INTEGER:
                value = Integer.valueOf(string);
                break;
            case SpInputType.STRING:
                value = String.valueOf(string);
                break;
            case SpInputType.LONG:
                value = Long.valueOf(string);
                break;
        }

    }

}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/fileexplorer/SpBean.kt
package com.didichuxing.doraemonkit.kit.fileexplorer

import com.didichuxing.doraemonkit.constant.SpInputType

/**
 * @author lostjobs created on 2020/6/14
 */
class SpBean(val key: String, var value: Any, val cls: Class<*>) {
    constructor(key: String, value: Any) : this(key, value, value::class.java)

    fun toDefaultClass(string: String): Any {
        this.value = when(cls.simpleName){
            SpInputType.BOOLEAN -> string.toBoolean()
            SpInputType.INTEGER -> string.toInt()
            SpInputType.LONG -> string.toLong()
            SpInputType.FLOAT -> string.toFloat()
            else -> string
        }
        return value
    }
}